pkg/asset/installconfig/ssh.go

```diff
@@ -39,32 +53,39 @@ func (a *sshPublicKey) Generate(map[asset.Asset]*asset.State) (state *asset.Stat
 		}, nil
 	}
 
-	pubKeys := map[string][]byte{
-		none: {},
-	}
-	home := os.Getenv("HOME")
-	if home != "" {
-		paths, err := filepath.Glob(filepath.Join(home, ".ssh", "*.pub"))
+	pubKeys := map[string][]byte{}
+	if path, ok := os.LookupEnv("OPENSHIFT_INSTALL_SSH_PUB_KEY_PATH"); ok {
+		key, err := readSSHKey(path)
 		if err != nil {
 			return nil, err
 		}
-
-		for _, path := range paths {
-			pubKeyBytes, err := ioutil.ReadFile(path)
+		pubKeys[path] = key
+	} else {
+		pubKeys[none] = []byte{}
+		home := os.Getenv("HOME")
+		if home != "" {
+			paths, err := filepath.Glob(filepath.Join(home, ".ssh", "*.pub"))
 			if err != nil {
-				continue
+				return nil, err
 			}
-			pubKey := string(pubKeyBytes)
-
-			err = validate.OpenSSHPublicKey(pubKey)
-			if err != nil {
-				continue
+			for _, path := range paths {
+				key, err := readSSHKey(path)
+				if err != nil {
+					continue
+				}
+				pubKeys[path] = key
 			}
-
-			pubKeys[path] = pubKeyBytes
 		}
 	}
 
+	if len(pubKeys) == 1 {
```

Comment:
We still need to allow the user to opt out of using a key at all.

Reply:
> We still need to allow the user to opt out of using a key at all.

I think we do.  Some cases:

* You set `OPENSHIFT_INSTALL_SSH_PUB_KEY` and exit [up here][1].
* You set `OPENSHIFT_INSTALL_SSH_PUB_KEY_PATH`, [get that single entry in your `pubKeys`][2], and exit here without a prompt.
* You set no variables, so you [get the `<none>` choice injected][3].  Then...
    * Go finds no valid pubkeys in `~/.ssh/*.pub`, and you exit here with the none choice selected.
    * Go finds some valid pubkeys in `~/.ssh/*.pub`, and you pass by here and fall down to the interactive prompt.

[1]: https://github.com/openshift/installer/pull/351/files#diff-0fa610c35b30a72dea1ff2a8c0b1e2e9R49
[2]: https://github.com/openshift/installer/pull/351/files#diff-0fa610c35b30a72dea1ff2a8c0b1e2e9R62
[3]: https://github.com/openshift/installer/pull/351/files#diff-0fa610c35b30a72dea1ff2a8c0b1e2e9R64

Reply:
Ah, that seems okay then.